src/app/common/utils.ts

```diff
@@ -254,3 +254,18 @@ type WhenPageModeMap<T> = Record<PageMode, T>;
 export function whenPageMode<T>(pageModeMap: WhenPageModeMap<T>) {
   return pageModeMap[pageMode];
 }
+interface WhenNetworkMap<T> {
+  [DefaultNetworkIds.mainnet]: T;
+  [DefaultNetworkIds.testnet]: T;
+  [DefaultNetworkIds.devnet]: T;
+}
```

Comment:
Think this is the same as `type WhenNetworkMap<T> = Record<DefaultNetworkIds, T>`, but both work.

I can't help but think we might start confusing two data types, and wonder if you have any ideas how we might differentiate:

```ts
// The type of networks available on Stacks blockchain
type TypesOfStacksNetworks = 'mainnet' | 'testnet'; // we have `whenChainId` already for this

// Default envs
type DefaultNetworkConfigurationsAvailableInWallet = 'mainnet' | 'testnet' | 'devnet';
```

Do we consider `devnet` a network? Is it accurate to call it a network, or is it just a configured testnet that points to localhost?

Reply:
I'm not thinking about bitcoin vs stacks.

There are two concepts here, and we're calling them both network, e.g. naming of `whenNetwork`, which is confusing. 

Devnet _is_ a testnet, and we have it in the network list as a testnet that points to localhost. Consider code below added here:

```ts
  const baseUrl = whenNetwork(network.id as DefaultNetworkNames)({
    [DefaultNetworkIds.mainnet]: BITCOIN_API_BASE_URL_MAINNET,
    [DefaultNetworkIds.testnet]: BITCOIN_API_BASE_URL_TESTNET,
    [DefaultNetworkIds.devnet]: BITCOIN_API_BASE_URL_TESTNET,
  });
```

Questions: 
- What if we add a new network of our own, which network do we point to?
- Why do we repeat the definition of a testnet url for devnet? If each `DefaultNetworkIds` has a corresponding "network" of either `mainnet | testnet`

Reply:
https://github.com/stacks-network/stacks-blockchain/blob/3e8a91733d673fa43fd653e2cc29213362da5560/src/core/mod.rs#L60-L61

Reply:
Yeah, I get your point now. I think we did refer to one of them as `NetworkModes` before, let me check. Let me step back and rethink this so makes more sense. Thanks for calling it out. 👍